elfloader.c

```diff
@@ -94,5 +99,10 @@ int main(int argc, char **argv)
     printf("Binary entrypoint is %x; invoking %p\n", ctx.ehdr.e_entry, ep);
```

Comment:
You changed this one back to
```
printf("Binary entrypoint is %x; invoking %p\n", ctx.ehdr.e_entry, ep);
```

Given `e_entry` is an `Elf_Addr`, it was wrong originally as `%x` and `%lx` is also wrong as `Elf_Addr` can be `int`, `long` or `long long` depending upon the platform

Given size wise `Elf_Addr` == `uintptr_t`, the correct line should be

```
printf("Binary entrypoint is %" PRIxPTR "; invoking %p\n", (uintptr_t) ctx.ehdr.e_entry, ep);
```

Otherwise, looks good